streamingpro-core/src/main/java/tech/mlsql/dsl/adaptor/DslAdaptor.scala

```diff
@@ -77,22 +78,22 @@ trait DslTool {
         split(",").map(_.trim).filterNot(_.isEmpty)
     }
 
-    val currentSchema = new URI(path).getScheme
-    if (filePathSchemas.contains(currentSchema)) {
-      return path
-    }
-
     val newPath = cleanStr(path)
     if (prefix.isEmpty) return newPath
 
-    if (path.contains("..")) {
+    if (newPath.contains("..")) {
       throw new RuntimeException("path should not contains ..")
     }
-    if (path.startsWith("/")) {
-      return prefix + path.substring(1, path.length)
+
+    if (newPath.startsWith("/")) {
+      return prefix + newPath.substring(1, newPath.length)
     }
-    return prefix + newPath
 
+    val currentSchema = extractSchema(path)
+    if (currentSchema.isEmpty || filePathSchemas.contains(currentSchema.get)) {
```

Comment:
The logical is not right. when schema is exits and is in  the pre-configured schema list , then we should let it go. Otherwise we should add prefix.